### PR TITLE
Feature/docker

### PR DESCRIPTION
**Pull request by dhsathiya**
Created this docker file because I have a lot of projects which are using GitHub Actions for CI/CD.
The image is heavy in size ~500MB but works like a charm and saves a lot time.

Tested on my Ubuntu 18.04 machine.
Using : 2019-08-06@20-47 Release.

--- 
Update : 
size reduced to 128MB

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM debian
+MAINTAINER devarshisathiya5@gmail.com
+RUN apt update 
+RUN apt install wget -y
```
Comment (ScottBrenner) [inline docker/Dockerfile]: Best to combine `update` & `install` steps in a single layer https://docs.docker.com/v17.09/engine/userguide/eng-image/dockerfile_best-practices/#run

**Comment by wei** [comment]
Why is go environment being installed when ready built binary is being downloaded?

File: docker/Dockerfile (hunk 1 above)
Comment (dhsathiya) [reply docker/Dockerfile]: I updated it. Also removed the apt list after the operations.
Thanks for noticing this and mentioning. I never got enough time to take look at best practices.

**Comment by dhsathiya** [comment]
@wei Thank you for noticing that I was adding Go environment unnecessarily.
I built this in extreme hurry and completely ignored the fact. :disappointed_relieved:

**Comment by dhsathiya** [comment]
@ScottBrenner I am using a specific version of the executable in the dockerfile
```
wget https://github.com/actions/migrate/releases/download/2019-08-06%4020-47/migrate-actions-linux.tar
```
Should I change it to latest ?
```
wget https://github.com/actions/migrate/releases/latest/download/migrate-actions-linux.tar
```

File: docker/Dockerfile (hunk 1 above)
Comment (ScottBrenner) [reply docker/Dockerfile]: > Note: The official Debian and Ubuntu images automatically run `apt-get clean`, so explicit invocation is not required.

**Comment by ScottBrenner** [comment]
Yes, I would recommend installing the latest release.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM debian
```
Comment (ScottBrenner) [inline docker/Dockerfile]: Any reason to use `debian` as the base image? Recommend Alpine Linux instead.

Comment (dhsathiya) [reply docker/Dockerfile]: Yes. I agree and I tried with `alpine`, `minideb` and `golang:alpine` as well.
I was getting these errors.
```
/entrypoint.sh: ./migrate-actions: line 1: ELF: not found
/entrypoint.sh: ./migrate-actions: line 2: @%: not found
/entrypoint.sh: ./migrate-actions: line 3: syntax error: unexpected "!"
```
After doing Google search I came to know that some libraries or `.so` files were missing.
I tried to install dependencies and what happened was, the errors persisted and image size increased more than 128MB.

Here is the basic Dockerfile of alpine that I used. I removed other dependency installations for now. https://gist.github.com/dhsathiya/987faca429dd4319bbf3c25272c6272d

File: docker/Dockerfile (hunk 1 above)
Comment (dhsathiya) [reply docker/Dockerfile]: I just checked that without `rm -rf /var/lib/apt/lists/*` the image size increases ~20MB.
I am adding it back. :slightly_smiling_face:

**Comment by dhsathiya** [comment]
I just realized that we don't need docker image for a 7MB executable.
I created the docker image to keep my system clean by not installing Go on my system.
It's better to create a shell script to run executable on all the cloned repositories.

Is there a option available in the executable to pass `path` as argument ?

What you say?

**Comment by wei** [comment]
You can put it in your bin folder or use the full path

**Comment by timruffles** [comment]
Thanks for this @dhsathiya. It'd certainly be simple to add a path, however, I'm not sure of the use-case for automating this conversion? Seems like you'd usually want to glance over the conversion anyhow (migrate over comments, parallel workflows etc).

**Comment by timruffles** [comment]
I'll close this for now - @dhsathiya if you feel there's a use case for the `path` arg please open an issue (or PR 😁 )

**Pull request closed without merging**